include/deal.II/fe/fe_poly.h

```diff
@@ -243,8 +243,10 @@ class FE_Poly : public FiniteElement<dim, spacedim>
   {
     // generate a new data object and
     // initialize some fields
-    auto data         = std_cxx14::make_unique<InternalData>();
-    data->update_each = requires_update_flags(update_flags);
+    std::unique_ptr<typename FiniteElement<dim, spacedim>::InternalDataBase>
+          data_ptr   = std_cxx14::make_unique<InternalData>();
```

Comment:
This conversion might trip up old compilers. I don't remember if the original issue was due to the extra complexity in dealing with the `return` statement or not.

Reply:
That was at least my interpretation of the `move` instruction from the `auto` type to the base type. Thanks for fixing this issue.